libs/agno/agno/workflow/v2/steps.py

```diff
@@ -32,21 +36,169 @@ class Steps:
     # Steps to execute
     steps: WorkflowSteps
 
-    # Pipeline_name identification
+    # Pipeline identification
     name: Optional[str] = None
     description: Optional[str] = None
 
-    def __init__(
-        self, name: Optional[str] = None, description: Optional[str] = None, steps: Optional[List[Step]] = None
-    ):
+    def __init__(self, name: Optional[str] = None, description: Optional[str] = None, steps: Optional[List] = None):
         self.name = name
         self.description = description
         self.steps = steps if steps else []
 
+    def _prepare_steps(self):
+        """Prepare the steps for execution - mirrors workflow logic"""
+        from agno.agent.agent import Agent
+        from agno.team.team import Team
+        from agno.workflow.v2.condition import Condition
+        from agno.workflow.v2.loop import Loop
+        from agno.workflow.v2.parallel import Parallel
+        from agno.workflow.v2.router import Router
+        from agno.workflow.v2.step import Step
+
+        prepared_steps = []
+        for step in self.steps:
+            if isinstance(step, Callable):
+                prepared_steps.append(Step(name=step.__name__, description="User-defined callable step", executor=step))
+            elif isinstance(step, Agent):
+                prepared_steps.append(Step(name=step.name, description=step.description, agent=step))
+            elif isinstance(step, Team):
+                prepared_steps.append(Step(name=step.name, description=step.description, team=step))
+            elif isinstance(step, (Step, Steps, Loop, Parallel, Condition, Router)):
+                prepared_steps.append(step)
+            else:
+                raise ValueError(f"Invalid step type: {type(step).__name__}")
+
+        self.steps = prepared_steps
+
+    def _update_step_input_from_outputs(
+        self,
+        step_input: StepInput,
+        step_outputs: Union[StepOutput, List[StepOutput]],
+        steps_step_outputs: Optional[Dict[str, StepOutput]] = None,
+    ) -> StepInput:
+        """Helper to update step input from step outputs - mirrors Condition/Router logic"""
+        current_images = step_input.images or []
+        current_videos = step_input.videos or []
+        current_audio = step_input.audio or []
+
+        if isinstance(step_outputs, list):
+            all_images = sum([out.images or [] for out in step_outputs], [])
```

Comment:
sum? Is that the correct python?

Reply:
yes, its another way to concatenate lists in python